types/levelup/levelup-tests.ts

```diff
@@ -63,7 +63,7 @@ db.isOpen();
 // $ExpectType boolean
 db.isClosed();
 
-// $ExpectType "closed" | "open" | "opening" | "new" | "closing"
+// $ExpectType "closed" | "open" | "opening" | "new" | "closing" || "open" | "closed" | "opening" | "new" | "closing"
```

Comment:
typo? is this needed for tests to pass?

Reply:
Union ordering is based on internal type IDs, and the type IDs are generated in an arbitrary order depending on the code that's loaded and the TS version. In this case, the undici types bring in more strings that node didn't have before, which means that the IDs shifted around and so these types have to be fixed.

Reply:
Isn't it problem with `$ExpectType`? To me ordering doesn't matter in string union.

Reply:
One could make the argument that it's a problem with ExpectType, or a problem with TypeScript, or both. It's just that (IIRC) nobody's figured out a way to improve this without negatively affecting performance. Until then, this really doesn't happen that often and is pretty easy to work around.

Reply:
Feel free to vote on / follow https://github.com/microsoft/TypeScript/issues/17944.  It looks like https://github.com/microsoft/dtslint/issues/191 wasn't resolved before that repo was closed.  They're tracking at https://github.com/microsoft/DefinitelyTyped-tools now but unfortunately they don't seem to have cloned over the open issues from the old one.  Might not hurt to open a new issue and link back to the unresolved one?

Reply:
To be clear, nobody _likes_ to have to do this; I don't intend for my comment to be some sort of official "as designed" status 😅

Thanks for linking that thread, which I entirely forgot to find and reread...